README.md

```diff
@@ -261,6 +261,52 @@ are not there.
 
 This makes laziness work right.
 
+### handling end, error, and abort.
+
+in pull streams, any part of the stream (source, sink, or through)
+may terminate the stream. (this is the case with node streams too,
+but it's not handled well).
+
+#### source: end, error
+
+A source may end (`cb(true)` after read) or error (`cb(error)` after read)
+After ending, the source *must* never `cb(null, data)`
+
+#### sink: abort
+
+Sinks do not normally end the stream, but if they decide they do
+not need any more data they may "abort" the source by calling `read(true, cb)`.
+A abort (`read(true, cb)`) may be called before a preceding read call
```

Comment:
Is the code example actually acceptable behaviour? Is it what you meant? "before a preceding read" confuses me. (do you mean "_after_ a preceding read"?